_docs/developer/developing_the_php_site/modules.md

```diff
@@ -0,0 +1,22 @@
+---
+title: ES6 Modules and Typescript
+---
+
+[ES6 Modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) allow JavaScript code to be split up
+into "modules" that can allow other modules to import functions and variables. Modules in Submitty are loaded the 
+same way as normal JavaScript files using the PHP function `$this->core->getOutput()->addInternalJs(...)`. Unlike normal 
+JavaScript, modules are "bundled" together by esbuild. This process takes all files imported and creates a single, 
+minified JavaScript file that is served to the user.
+
```

Comment:
Let's add a note here like:

> All new JavaScript modules should written using TypeScript.

There's really no reason to allow a mix going forward, and the type safety of TS is really nice, especially given that modules avoids issues of global scoping.